dask_expr/io/parquet.py

```diff
@@ -773,6 +773,19 @@ def approx_statistics(self) -> dict:
         stats = [_STATS_CACHE[tokenize(finfo)] for finfo in files_to_consider]
         return _combine_stats(stats)
 
+    @cached_property
+    def unique_partition_mapping_columns(self):
+        # TODO This is a poor name
+        if self.calculate_divisions:
+            return {
+                col
+                for col in self.columns
+                if _divisions_from_statistics(self.aggregated_statistics, col)[1]
+                is not None
+            }
+        else:
+            return set()
```

Comment:
Note that this information is not sufficient to perform merges. It is sufficient to perform trivial groupby+apply operations but not to join two frames unless the statistics between the two frames are identical. 
It looks like your merge implementation doesn't distinguish this. We may need to propagate two distinct properties

Reply:
I removed this for now, since that's not relevant at the moment for benchmarks. But yeah I think we have to use a different property then